src/metatensor/models/utils/data/writers/xyz.py

```diff
@@ -75,50 +66,47 @@ def write_xyz(
 
             for gradient_name, gradient_block in block.gradients():
                 # here, we assume that gradients are always an array, and never a scalar
-                if capabilities.outputs[target_name].quantity == "energy":
-                    if gradient_name == "positions":
-                        if only_one_energy:
-                            name_for_saving = "forces"
-                        else:
-                            name_for_saving = f"forces[{target_name}]"
-                        arrays[name_for_saving] = (
-                            # squeeze the property dimension
-                            -gradient_block.values.detach()
-                            .cpu()
-                            .squeeze(-1)
-                            .numpy()
-                        )
-                    elif gradient_name == "strain":
-                        strain_derivatives = (
-                            # squeeze the property dimension
-                            gradient_block.values.detach()
-                            .cpu()
-                            .squeeze(-1)
-                            .numpy()
+                internal_name = f"{target_name}_{gradient_name}_gradients"
+                external_name = to_external_name(internal_name, capabilities.outputs)
+
+                if "forces" in external_name:
+                    arrays[external_name] = (
+                        # squeeze the property dimension
+                        -gradient_block.values.detach()
+                        .cpu()
+                        .squeeze(-1)
+                        .numpy()
+                    )
+                elif "virial" in external_name:
+                    # here, we write the virials as stresses, but we could write them
+                    # as virials as well just to make sure
+                    external_name = external_name.replace("virial", "stress")
```

Comment:
Maybe we should inform the user here that we rename something?

Reply:
Yes, I'm not sure how to handle this. The issue is that we convert both virials and stresses to the same thing internally (and rightfully so, in my opinion), but then we lose the information of what the input was initially. In the end we have to pick one of the two

Reply:
You mean we have no idea in the end what in initial input was?

Basically, if in the end the user does not see any changes we also don't have to warn.

Reply:
Ok! We decided on writing both to the output in that case